### PR TITLE
Header responsible for design, GlobalNavigation refactoring

### LINKED ISSUE #16: GlobalNavigation should work independent of children
# Description

During the redesign yesterday I introduced some rough dependencies to the GlobalNavigation.

For instance it currently requires 'NavigationItems' as children to work at all, but the placed checks are not appropriate for that.

Further, the coloring of the button for the mobile drawer is colored in this component itself, but should be configurable.

# Tasks

- [ ] Remove dependency on 'NavigationItems' for 'GlobalNavigation'
- [ ] Introduce appropriate styling properties for the component

### PR DESCRIPTION
**Pull request by ghost**
This pull request closes #16 by removing the dependency to `NavigationItem` from the `GlobalNavigation`. Further, it introduces styling options on the NavigationItem as well as the `GlobalNavigation` to make everything configurable from the header component.

### CODE REVIEW

**Review by lorenzo-w — CHANGES_REQUESTED** [review]
I like it a lot better already, but tell me again: What purpose does NavigationItem serve? I mean: Why could I not just style a Link or Box any way I want and give it directly to GlobalNavigation? 
Since Header already chooses the color values, why not go one step further and have it set them directly via a standardized CSS interface, instead of creating yet another level of indirection?
IMO the only thing that does warrant a custom prop is the background color of the Drawer. 

**Bottom line: We should avoid writing custom props for styling features as much as possible, as each of those will be yet another piece of API to learn for any new contributor**. Use CSS (in our case CSS-in-JS) and well-known component interfaces like those of MUI as much as you can.

**Comment by thsmdt** [comment]
I realize that `NavigationItem` has little functionality on its own aside from encapsulating styles away. From my perspective, creating a component is not just about broad reuse, such as our `QHeading` component, but also about isolation and single-responsibility.

If its entire code was part of the `Header`, granted, we had one less indirection and higher flexibility in styling [0]. However, the header would be bloated with more stuff to handle, providing the links and typography themselves, thus having a higher complexity and in my opinion violating single-responsibility. Further, a separate component, like NavigationItem, helps signal to all developers what this part is doing by name, compared to inline generic ones.

To keep the components customizable they need styling options in some way and I see your point about the added API here. It is a trade-off where I personally prefer smaller components with descriptive names. Nevertheless, if you feel strongly about it, let us remove NavigationItem entirely.

[0]: I am currently looking into improving this (e.g., `sx` support for custom components, passing `ResponsiveStyleValue` instead of strings).

**Review by lorenzo-w — APPROVED** [review]
